ambry-router/src/test/java/com.github.ambry.router/DeleteManagerTest.java

```diff
@@ -353,21 +328,19 @@ public void testVariousServerErrorCodesForThreeParallelism()
   public void testResponseTimeout()
       throws Exception {
     setServerResponse(false);
-    ServerErrorCode[] serverErrorCodes = new ServerErrorCode[9];
-    Arrays.fill(serverErrorCodes, ServerErrorCode.No_Error);
-    presetServerErrorCode(serverErrorCodes);
-    future = router.deleteBlob(blobIdString, new ClientCallback());
-    do {
-      // increment mock time
-      mockTime.sleep(1000);
-    } while (!operationCompleteLatch.await(10, TimeUnit.MILLISECONDS));
-    try {
-      future.get();
-      fail("Deletion should be unsuccessful. Exception is expected.");
-    } catch (Exception e) {
-      assertEquals("RouterErrorCode should be OperationTimedOut", RouterErrorCode.OperationTimedOut,
-          ((RouterException) e.getCause()).getErrorCode());
-    }
+    testWithErrorCodes(Collections.singletonMap(ServerErrorCode.No_Error, 9), serverLayout,
+        RouterErrorCode.OperationTimedOut, new ErrorCodeChecker() {
+          @Override
+          public void check(RouterErrorCode expectedError)
+              throws Exception {
+            future = router.deleteBlob(blobIdString, new ClientCallback());
+            do {
+              // increment mock time
+              mockTime.sleep(1000);
+            } while (!operationCompleteLatch.await(10, TimeUnit.MILLISECONDS));
```

Comment:
@cgtz 
The timeout should be `RouterConfig.routerRequestTimeoutMs` instead of `RouterConfig.routerConnectionCheckoutTimeoutMs`. The former one is set 2000ms. Theoretically the while loop (`mockTime.sleep(100)`) should be executed 4 times:
1st: time = 0, less than 2000
2nd: time =1000, less than 2000
3rd: time=2000, equal to 2000 but the thread to complete the operation did not get scheduled to complete it.
4th: time=3000, operation complete, jump out of the while loop.

However, if the timeout for operationCompleteLatch.await() is too small, it may run more loops since the thread to complete the operation does not get a chance to be scheduled. I think this was the original reason to set the timeout to be 10. However, it should not hurt to make it 1ms, which will run more while loops but spend less time for each loop.

My suggestion is that change this time to 1ms, and then run the test multiple times to ensure no issue.